Pod/Classes/Categories/UIImage+DecodedImage.m

```diff
@@ -69,22 +83,91 @@ + (UIImage *)pin_decodedImageWithCGImageRef:(CGImageRef)imageRef
     }
     
     CGSize imageSize = CGSizeMake(CGImageGetWidth(imageRef), CGImageGetHeight(imageRef));
-    UIGraphicsBeginImageContextWithOptions(imageSize, opaque, 1.0);
-    CGContextRef ctx = UIGraphicsGetCurrentContext();
+    
+    CGBitmapInfo info = opaque ? (kCGImageAlphaNoneSkipFirst | kCGBitmapByteOrder32Host) : (kCGImageAlphaPremultipliedFirst | kCGBitmapByteOrder32Host);
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    
+    //Use UIGraphicsBeginImageContext parameters from docs: https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIKitFunctionReference/#//apple_ref/c/func/UIGraphicsBeginImageContextWithOptions
+    CGContextRef ctx = CGBitmapContextCreate(NULL, imageSize.width, imageSize.height,
+                                             8,
+                                             0,
+                                             colorspace,
+                                             info);
+    
+    CGColorSpaceRelease(colorspace);
+    
     UIImage *decodedImage = nil;
     if (ctx) {
-        CGContextScaleCTM(ctx, 1.0, -1.0);
-        CGContextTranslateCTM(ctx, 0.0, -imageSize.height);
         CGContextDrawImage(ctx, CGRectMake(0, 0, imageSize.width, imageSize.height), imageRef);
-        decodedImage = UIGraphicsGetImageFromCurrentImageContext();
-        UIGraphicsEndImageContext();
+        
+        CGImageRef newImage = CGBitmapContextCreateImage(ctx);
+        
+        decodedImage = [UIImage imageWithCGImage:newImage scale:1.0 orientation:orientation];
+        
+        CGImageRelease(newImage);
+        
     } else {
-        decodedImage = [UIImage imageWithCGImage:imageRef];
+        decodedImage = [UIImage imageWithCGImage:imageRef scale:1.0 orientation:orientation];
     }
     
     return decodedImage;
 }
 
++(UIImageOrientation) pin_UIImageOrienationFromImageSource:(CGImageSourceRef)imageSourceRef {
```

Comment:
I would consider making this a static helper function in the implementation file.